File: leetcode/958.二叉树的完全性检验.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
type Node struct  {
    node *TreeNode
    val int
}
func isCompleteTree(root *TreeNode) bool {
    //bfs检验是否是完全二叉树
    nodes := []Node{Node{root, 1}}
    i := 0
    for i < len(nodes) {
        node, v := nodes[i].node, nodes[i].val
        i += 1
        if node != nil {
            nodes = append(nodes, Node{node.Left, 2 * v})
            nodes = append(nodes, Node{node.Right, 2 * v + 1})
        }
    }
    fmt.Print(nodes)
    return nodes[len(nodes) - 1].val == len(nodes)
}

File: leetcode/958.二叉树的完全性检验.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:
        cnt,mx=0,0
        def dfs(root,k):
            nonlocal cnt,mx
            if not root:
                return 
            cnt+=1
            mx=max(mx,k)
            dfs(root.left,2*k)
            dfs(root.right,2*k+1)
        dfs(root,1)
        return cnt==mx   
        
